File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/property/PropertyAssignment.js
define([ "stjs", "util/Exceptions" ], function(stjs, exceptions) {
	var PropertyAssignment = function() {
	};
	stjs.extend(PropertyAssignment, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.property.PropertyAssignment';
		constructor.serialVersionUID = 1;
		prototype.fetchOptions = null;
		prototype.permId = null;
		prototype.section = null;
		prototype.ordinal = null;
		prototype.entityType = null;
		prototype.propertyType = null;
		prototype.mandatory = null;
		prototype.showInEditView = null;
		prototype.showRawValueInForms = null;
		prototype.semanticAnnotations = null;
		prototype.registrator = null;
		prototype.registrationDate = null;

		prototype.getFetchOptions = function() {
			return this.fetchOptions;
		};
		prototype.setFetchOptions = function(fetchOptions) {
			this.fetchOptions = fetchOptions;
		};
		prototype.getPermId = function() {
			return this.permId;
		};
		prototype.setPermId = function(permId) {
			this.permId = permId;
		};
		prototype.getSection = function() {
			return this.section;
		};
		prototype.setSection = function(section) {
			this.section = section;
		};
		prototype.getOrdinal = function() {
			return this.ordinal;
		};
		prototype.setOrdinal = function(ordinal) {
			this.ordinal = ordinal;
		};
		prototype.getEntityType = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasEntityType()) {
				return this.entityType;
			} else {
				throw new exceptions.NotFetchedException("Entity type has not been fetched.");
			}
		};
		prototype.setEntityType = function(entityType) {
			this.entityType = entityType;
		};
		prototype.getPropertyType = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasPropertyType()) {
				return this.propertyType;
			} else {
				throw new exceptions.NotFetchedException("Property type has not been fetched.");
			}
		};
		prototype.setPropertyType = function(propertyType) {
			this.propertyType = propertyType;
		};
		prototype.isMandatory = function() {
			return this.mandatory;
		};
		prototype.setMandatory = function(mandatory) {
			this.mandatory = mandatory;
		};
		prototype.isShowInEditView = function() {
			return this.showInEditView;
		};
		prototype.setShowInEditView = function(showInEditView) {
			this.showInEditView = showInEditView;
		};
		prototype.isShowRawValueInForms = function() {
			return this.showRawValueInForms;
		};
		prototype.setShowRawValueInForms = function(showRawValueInForms) {
			this.showRawValueInForms = showRawValueInForms;
		};
		prototype.getSemanticAnnotations = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasSemanticAnnotations()) {
				return this.semanticAnnotations;
			} else {
				throw new exceptions.NotFetchedException("Semantic annotations have not been fetched.");
			}
		};
		prototype.setSemanticAnnotations = function(semanticAnnotations) {
			this.semanticAnnotations = semanticAnnotations;
		};
		prototype.getRegistrator = function() {
			if (this.getFetchOptions() && this.getFetchOptions().hasRegistrator()) {
				return this.registrator;
			} else {
				throw new exceptions.NotFetchedException("Registrator has not been fetched.");
			}
		};
		prototype.setRegistrator = function(registrator) {
			this.registrator = registrator;
		};
		prototype.getRegistrationDate = function() {
			return this.registrationDate;
		};
		prototype.setRegistrationDate = function(registrationDate) {
			this.registrationDate = registrationDate;
		};
	}, {
		fetchOptions : "PropertyAssignmentFetchOptions",
		entityType : "IEntityType",
		propertyType : "PropertyType",
		registrator : "Person",
		registrationDate : "Date"
	});
	return PropertyAssignment;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/property/PropertyAssignment.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.property;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IEntityType;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IPermIdHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IPropertyTypeHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IRegistrationDateHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.IRegistratorHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.Person;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.PropertyType;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.fetchoptions.PropertyAssignmentFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.id.PropertyAssignmentPermId;
import ch.ethz.sis.openbis.generic.asapi.v3.exceptions.NotFetchedException;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;
import java.util.Date;

/*
 * Class automatically generated with DtoGenerator
 */
@JsonObject("as.dto.property.PropertyAssignment")
public class PropertyAssignment implements Serializable, IPermIdHolder, IPropertyTypeHolder, IRegistrationDateHolder, IRegistratorHolder
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private PropertyAssignmentFetchOptions fetchOptions;

    @JsonProperty
    private PropertyAssignmentPermId permId;

    @JsonProperty
    private String section;

    @JsonProperty
    private Integer ordinal;

    @JsonProperty
    private IEntityType entityType;

    @JsonProperty
    private PropertyType propertyType;

    @JsonProperty
    private Boolean mandatory;

    @JsonProperty
    private Boolean showInEditView;

    @JsonProperty
    private Boolean showRawValueInForms;

    @JsonProperty
    private Person registrator;

    @JsonProperty
    private Date registrationDate;

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public PropertyAssignmentFetchOptions getFetchOptions()
    {
        return fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public void setFetchOptions(PropertyAssignmentFetchOptions fetchOptions)
    {
        this.fetchOptions = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public PropertyAssignmentPermId getPermId()
    {
        return permId;
    }

    // Method automatically generated with DtoGenerator
    public void setPermId(PropertyAssignmentPermId permId)
    {
        this.permId = permId;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public String getSection()
    {
        return section;
    }

    // Method automatically generated with DtoGenerator
    public void setSection(String section)
    {
        this.section = section;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Integer getOrdinal()
    {
        return ordinal;
    }

    // Method automatically generated with DtoGenerator
    public void setOrdinal(Integer ordinal)
    {
        this.ordinal = ordinal;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public IEntityType getEntityType()
    {
        if (getFetchOptions() != null && getFetchOptions().hasEntityType())
        {
            return entityType;
        }
        else
        {
            throw new NotFetchedException("Entity type has not been fetched.");
        }
    }

    // Method automatically generated with DtoGenerator
    public void setEntityType(IEntityType entityType)
    {
        this.entityType = entityType;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public PropertyType getPropertyType()
    {
        if (getFetchOptions() != null && getFetchOptions().hasPropertyType())
        {
            return propertyType;
        }
        else
        {
            throw new NotFetchedException("Property type has not been fetched.");
        }
    }

    // Method automatically generated with DtoGenerator
    public void setPropertyType(PropertyType propertyType)
    {
        this.propertyType = propertyType;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Boolean isMandatory()
    {
        return mandatory;
    }

    // Method automatically generated with DtoGenerator
    public void setMandatory(Boolean mandatory)
    {
        this.mandatory = mandatory;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Boolean isShowInEditView()
    {
        return showInEditView;
    }

    // Method automatically generated with DtoGenerator
    public void setShowInEditView(Boolean showInEditView)
    {
        this.showInEditView = showInEditView;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    public Boolean isShowRawValueInForms()
    {
        return showRawValueInForms;
    }

    // Method automatically generated with DtoGenerator
    public void setShowRawValueInForms(Boolean showRawValueInForms)
    {
        this.showRawValueInForms = showRawValueInForms;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public Person getRegistrator()
    {
        if (getFetchOptions() != null && getFetchOptions().hasRegistrator())
        {
            return registrator;
        }
        else
        {
            throw new NotFetchedException("Registrator has not been fetched.");
        }
    }

    // Method automatically generated with DtoGenerator
    public void setRegistrator(Person registrator)
    {
        this.registrator = registrator;
    }

    // Method automatically generated with DtoGenerator
    @JsonIgnore
    @Override
    public Date getRegistrationDate()
    {
        return registrationDate;
    }

    // Method automatically generated with DtoGenerator
    public void setRegistrationDate(Date registrationDate)
    {
        this.registrationDate = registrationDate;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public String toString()
    {
        return "PropertyAssignment entity type: " + (entityType != null ? entityType.getCode() : null) + ", property type: " + (propertyType != null ? propertyType.getCode() : null) + ", mandatory: " + mandatory;
    }

}
